File: src/main/java/pro/tuscan/config/MongoConfiguration.java
package pro.tuscan.config;

import com.mongodb.MongoClientOptions;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import pro.tuscan.config.properties.MongoClientProperties;

import static com.mongodb.WriteConcern.ACKNOWLEDGED;

@Configuration
public class MongoConfiguration {

    private final MongoClientProperties properties;

    public MongoConfiguration(MongoClientProperties properties) {
        this.properties = properties;
    }

    @Bean
    MongoClientOptions mongoClientOptions() {
        return MongoClientOptions.builder()
                .socketTimeout(properties.getSocketTimeout())
                .connectionsPerHost(properties.getConnectionsPerHost())
                .maxConnectionIdleTime(properties.getMaxConnectionIdleTime())
                .writeConcern(ACKNOWLEDGED)
                .build();
    }
}

File: src/main/java/pro/tuscan/config/MongoConfiguration.kt
package pro.tuscan.config

import com.mongodb.MongoClientOptions
import com.mongodb.WriteConcern
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import pro.tuscan.config.properties.MongoClientProperties

@Configuration
class MongoConfiguration(private val properties: MongoClientProperties) {

    @Bean
    fun mongoClientOptions(): MongoClientOptions =
            MongoClientOptions.builder()
                    .socketTimeout(properties.socketTimeout)
                    .connectionsPerHost(properties.connectionsPerHost)
                    .maxConnectionIdleTime(properties.maxConnectionIdleTime)
                    .writeConcern(WriteConcern.ACKNOWLEDGED)
                    .build()
}
